docs/src/tutorial/cython_tutorial.rst

```diff
@@ -132,27 +132,83 @@ them as a Python list.
     :linenos:
 
 You'll see that it starts out just like a normal Python function definition,
-except that the parameter ``kmax`` is declared to be of type ``int`` . This
+except that the parameter ``nb_primes`` is declared to be of type ``int`` . This
 means that the object passed will be converted to a C integer (or a
 ``TypeError.`` will be raised if it can't be).
 
+Now, let's dig into the core of the function::
+
+    cdef int n, i, len_p
+    cdef int p[1000]
+
 Lines 2 and 3 use the ``cdef`` statement to define some local C variables.
-Line 4 creates a Python list which will be used to return the result. You'll
-notice that this is done exactly the same way it would be in Python. Because
-the variable result hasn't been given a type, it is assumed to hold a Python
-object.
+The result is put in ``p``, it will be converted to a python list at the end
```

Comment:
It's not clear here what "p" is, so maybe:
"The result is stored in the statically allocated C array ``p`` during processing, and will be copied into a Python list at the end ..."

Reply:
Hmm, although - "statically" is actually wrong, it's "stack allocated". But it's a bit difficult to be both accurate here (for someone who knows C) and easily understandable (for someone who does not know what stack allocation is)...

Reply:
We could just say 'The result is stored in the C array p during processing, and will be copied into a Python list at the end'. It's not wrong and C users will discover later on that it was allocated on the stack.

Reply:
With the caveat that stack allocation of larger chunks of data is something that should generally be avoided. The example is using up 4K of stack space on most systems, that's a lot. New users who don't know about this could easily learn bad habits from this.

I'm aware that this is what the example always did, so you're not introducing a new problem here. Maybe adding a note on the memory behaviour and linking to the memory allocation chapter would help.

Reply:
Ok I'll do that.